qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/gsas2/presenter.py

```diff
@@ -55,8 +56,16 @@ def get_limits_if_same_load_parameters(self):
             return None
         current_limits = self.view.get_x_limits_from_line_edits()
         if current_limits:
-            if current_limits != self.view.initial_x_limits:
-                return sorted(current_limits)
+            # Check current_limits != self.view.initial_x_limits:
+            if not functools.reduce(
+                lambda p, q: p and q,
+                map(
+                    lambda cur_limit, init_limit: round(float(cur_limit[0]), 2) == round(init_limit, 2),
```

Comment:
It's v minor, but I think this is quite complicated/difficult to read. I think it aims to loop over tuples of limits and check they have not changed, unless I've mi-understood? Could you do this by making a numpy array and using `np.allclose` or similar? If it's more complicate don't worry, I'm happy to merge as is.

Quick question, why is it only the first element `cur_limit[0]` that is being compared 0 is this just for the lower limit?

Reply:
Basically, this checks whether `current_limits != self.view.initial_x_limits` and it had been coded like that as it is which will not work as expected since values inside current_limits are rounded to 2 decimals and hence the new code was added to do the comparison.
I have made it simpler with np.allclose. 

The reason for using 0 index here is because the array returned by `self.view.get_x_limits_from_line_edits()` is 2-dimensional like [[xmin], [xmax]].